specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-06-28-preview/confidentialledger.json

```diff
@@ -0,0 +1,830 @@
+        }
+      }
+    },
+    "AADBasedSecurityPrincipal": {
+      "type": "object",
+      "description": "AAD based security principal with associated Ledger RoleName",
+      "properties": {
+        "principalId": {
+          "description": "UUID/GUID based Principal Id of the Security Principal",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "UUID/GUID based Tenant Id of the Security Principal",
+          "type": "string"
+        },
+        "ledgerRoleName": {
+          "$ref": "#/definitions/LedgerRoleName"
+        }
+      }
+    },
+    "LedgerProperties": {
+      "type": "object",
+      "description": "Additional Confidential Ledger properties.",
+      "title": "LedgerProperties",
+      "properties": {
+        "ledgerName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Unique name for the Confidential Ledger."
+        },
+        "ledgerUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for calling Ledger Service."
+        },
+        "identityServiceUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for accessing network identity."
+        },
+        "ledgerInternalNamespace": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Internal namespace for the Ledger"
+        },
+        "runningState": {
+          "description": "Object representing RunningState for Ledger.",
+          "$ref": "#/definitions/RunningState"
+        },
+        "ledgerType": {
+          "description": "Type of Confidential Ledger",
+          "$ref": "#/definitions/ConfidentialLedgerType"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of Ledger Resource",
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "aadBasedSecurityPrincipals": {
+          "description": "Array of all AAD based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AADBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": [
+            "principalId"
+          ]
+        },
+        "certBasedSecurityPrincipals": {
+          "description": "Array of all cert based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CertBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "ConfidentialLedgerBackup": {
+      "description": "Object representing Backup properties of Confidential Ledger Resource.",
+      "type": "object",
+      "properties": {
+        "restoreRegion": {
+          "type": "string",
+          "readOnly": false,
+          "description": "The region where the backup of the Ledger Resource will eventually be restored to."
```

Comment:
> "The region where the backup of the Ledger Resource will eventually be restored to." [](http://example.com/codeflow?start=24&length=85)

do people need to predict the future? what if they change their mind?

Reply:
For our service, we need to prepare a few records in the future restore region when our resource is being backed up. This requires the user to specify which region they plan to restore their resource to if it is different from the original region and that is why this field is needed. I agree that it is not ideal but it is required for our service in order for customers to perform a cross-region restore.